crates/runtime/src/http/v1/responses.rs

```diff
@@ -0,0 +1,259 @@
+            serde_json::Value = "application/json",
+            example = json!({
+                "error": "An internal server error occurred while processing the chat completion."
+            })
+        )))
+    )
+))]
+pub(crate) async fn post(
+    Extension(llms): Extension<Arc<RwLock<LLMResponsesModelStore>>>,
+    headers: HeaderMap,
+    Json(req): Json<CreateResponse>,
+) -> Response {
+    let span = tracing::span!(
+        target: "task_history",
+        tracing::Level::INFO,
+        "ai_chat",
+        input = %serde_json::to_string(&req).unwrap_or_default()
+    );
+    span.in_scope(|| tracing::info!(target: "task_history", model = %req.model, "labels"));
+
+    override_task_history_with_traceparent(&span.clone(), &headers);
+
+    let span_clone = span.clone();
+    async move {
+        let model_id = req.model.clone();
+        let stream = req.stream.unwrap_or(false);
+
+        let Some(model) = llms.read().await.get(&model_id).cloned() else {
+            return (StatusCode::NOT_FOUND, format!("model '{model_id}' not found")).into_response();
+        };
+
+        if stream {
+            // Streaming response
+            create_response_sse_response(model, req, span_clone).await
+        } else {
+            // Non-streaming response
+            match model.responses_request(req).await {
+                Ok(response) => {
+                    let message = extract_text(&response);
+                    if !message.is_empty() {
+                        tracing::info!(target: "task_history", parent: &span_clone, captured_output = %message);
+                    }
+                    tracing::info!(target: "task_history", parent: &span_clone,  id = %response.id, "labels");
+
+                    Json(response).into_response()
+                }
+                Err(e) => {
+                    tracing::error!(target: "task_history", parent: &span_clone, "{e}");
+                    tracing::error!("Error from v1/responses: {e}");
+
+                    openai_error_to_response(e)
+                }
+            }
+        }
+    }
+    .instrument(span)
+    .await
+}
+
+fn to_openai_error_event(err: impl Into<String>) -> Event {
+    Event::default().event("error").data(
+        serde_json::to_string(&OpenaiErrorEvent::new(err))
+            .unwrap_or_else(|_| r#"{"error": "Failed to serialize error"}"#.to_string()),
+    )
+}
+
+/// Create a SSE [`axum::response::Response`] from a [`ResponseStream`].
+async fn create_response_sse_response(
+    model: Arc<dyn Responses>,
+    req: CreateResponse,
+    span: Span,
+) -> Response {
+    let mut strm: ResponseStream = match model.responses_stream(req).await {
+        Ok(stream) => stream,
+        Err(e) => {
+            tracing::error!(target: "task_history", parent: &span, "{e}");
+            tracing::error!("Error from v1/responses: {e}");
+            return openai_error_to_response(e);
+        }
+    };
+
+    let sse_stream = stream! {
```

Comment:
Rework this to not use the `stream!` macro - I think its fine for small amounts of code to use the `stream!` macro, but it breaks rust-analyzer from being able to provide useful hints while typing, and so I would avoid for large code blocks.